hatchways/team-pine
# Manage bookings integration

### What this PR does:
- This connects the front end of the Manage bookings page to the back end. All bookings are now retrieved via API calls to the request controller.
- Status can be changed via a dropdown menu for each booking. Calendar still displays dates based on upcoming bookings, but now only for accepted ones.

### Screenshots / Videos:
![image](https://user-images.githubusercontent.com/34494229/150230156-7ce59d9c-97e1-47f7-950d-98bf735147ff.png)

### Any information needed to test this feature:
- Start the server section of the app with appropriate database settings
- Make sure you have valid requests in your database. If you do not, you need:
1. At least two users, one who is the requester and one who is the sitter. You can register new users on the /register page.
2. A few requests, with one from a past date if you wish to see the past bookings section populated. You may also test without bookings to check how the page handles having no data. You may create requests through postman after logging in with a valid POST /auth/login request. As long as you get a cookie, you should then be able to create requests by calling POST at /requests with a valid body structure. Check the model to see how to structure the body.

- You will want to log in to a user who has requests as a sitter to test the status changing with the settings icon buttons.

### Any issues with the current functionality:
- The calendar does not update with the status changes until refresh, whereas the statuses themselves on the booking cards do.
- Users can click on dates with this view, which can cause confusion as to what dates their upcoming bookings are on (the styling is the same). However, they can only do this with one date.
- Their is no current distinction with whether the user is a owner or sitter, so you can navigate to the bookings page as an owner. It should not let you update status on the backend, however, even if it changes on the screen.
- There is currently no integration with profile pictures.

closes #31 

### Review thoughts:
- Bookings.tsx is getting pretty big, but I think it is still readable. Let me know if I am wrong!
